src/parse.ts

```diff
@@ -25,7 +25,20 @@ export function getCommand(
   if (!c)
     throw new UnsupportedCommand({ agent, command })
 
-  return c.replace('{0}', args.join(' ')).trim()
+  const quote = (arg: string) => {
+    const hasIfPresent = arg.startsWith('--if-present ')
+    const prefix = hasIfPresent ? '--if-present ' : ''
+    if (hasIfPresent)
+      arg = arg.slice('--if-present '.length)
```

Comment:
Why is this introduced here?

Reply:
https://github.com/antfu/ni/blob/d7951e7b5a5822a496c1f8c75c152c379bda4ab7/src/parse.ts#L59
Because `--if-present` has been inserted into the first parameter.
I tried to replace "args[0] = `--if-present ${args[0]}` " to `args.unshift('--if-present')`, but this could lead to other issues, perhaps you could make some overall adjustments